src/win/fs.c

```diff
@@ -2627,8 +2627,56 @@ static void fs__statfs(uv_fs_t* req) {
                              &bytes_per_sector,
                              &free_clusters,
                              &total_clusters)) {
-    SET_REQ_WIN32_ERROR(req, GetLastError());
-    return;
+    DWORD err;
+
+    err = GetLastError();
+    if (err == ERROR_DIRECTORY) {
```

Comment:
Maybe you could check `err != ERROR_DIRECTORY` here. The current `else` branch is a lot simpler and returns, so the `ERROR_DIRECTORY` handling branch could lose a level of indentation.